crates/burn-train/src/metric/classification.rs

```diff
@@ -1,3 +1,24 @@
+use std::num::NonZeroUsize;
+
+/// Necessary data for classification metrics.
+#[derive(Default)]
+pub struct ClassificationMetricConfig {
+    pub decision_rule: DecisionRule,
+    pub class_reduction: ClassReduction,
+}
+
+/// The prediction decision rule for classification metrics.
+pub enum DecisionRule {
+    Threshold(f64),
+    TopK(NonZeroUsize),
+}
```

Comment:
Ok! I think I've come around to the terminology behind this representation.

Just make sure to document both decision rules. Suggestions:

Top-k: Consider the prediction correct if the true label is within the top k predicted classes based on scores.
Threshold: Consider a class predicted if its probability exceeds the threshold.

Reply:
Great, I'm glad! Thanks for the sugestions, I changed `TopK`'s slightly. Let me know what you think.